File: ios/DataSources/MQTT/objective-c/MQTTDataRequest.m
//
//  MQTTDataRequest.m
//  MQTTMobile
//
//  Created by Stephen Schalkhauser on 1/7/18.
//  Copyright © 2018 P2UX, Inc. All rights reserved.
//

#import "MQTTDataRequest.h"

@implementation MQTTDataRequest

- (instancetype) initWithTopic:(NSString*)topic qos:(MQTTQosLevel)qos request:(NSString*)request callback:(P2UXCallbackResponse)callback
{
    self = [super init];
    if (self != nil)
    {
        _callback = callback;
        _request = request;
        _topic = topic;
        _qos = qos;
    }
    return self;
}

- (void) cleanup
{
    _callback = nil;
    _request = nil;
    _topic = nil;
    _requestSpec = nil;
    _schemaSpec = nil;
    _args = nil;
}

- (void) setRequestInstanceData:(NSDictionary*)requestSpec schemaSpec:(NSDictionary*)schemaSpec args:(NSDictionary*)args 
{
    _requestSpec = requestSpec;
    _schemaSpec = schemaSpec;
    _args = args;
}

- (BOOL) hasRequestData
{
    return _requestSpec != nil;
}
@end

File: ios/DataSources/MQTT/swift/MQTTDataRequest.swift
//
//  MQTTDataRequest.swift
//  MQTTMobile
//
//  Created by Stephen Schalkhauser on 1/9/18.
//  Copyright © 2018 P2UX, Inc. All rights reserved.
//

import MQTTFramework
import P2UXApp

class MQTTDataRequest {
    var request : String
    var requestSpec : NSDictionary?
    var schemaSpec : NSDictionary?
    var args : NSDictionary?
    var topic : String
    var qos : MQTTQosLevel
    var callback : P2UXCallbackResponse?
    
    var hasRequestData : Bool {
        get {
            return requestSpec != nil
        }
    }
    
    init(withTopic topic: String, qos: MQTTQosLevel, request: String, callback: @escaping P2UXCallbackResponse) {
        self.topic = topic
        self.qos = qos
        self.request = request
        self.callback = callback
    }

    init(withTopic topic: String, qos: MQTTQosLevel, request: String) {
        self.topic = topic
        self.qos = qos
        self.request = request
    }

    func setRequestInstanceData(requestSpec: NSDictionary, schemaSpec: NSDictionary, args: NSDictionary) {
        self.requestSpec = requestSpec
        self.schemaSpec = schemaSpec
        self.args = args
    }
    
}
